mslice/models/slice/mantid_slice_algorithm.py

```diff
@@ -82,6 +82,15 @@ def compute_symmetrised(self, scattering_data, boltzmann_dist, y_axis):
         negatives = scattering_data[len(energy_transfer):] * boltzmann_dist[len(energy_transfer):,None]
         return np.concatenate((scattering_data[:len(energy_transfer)], negatives))
 
+    def compute_gdos(self, scattering_data, boltzmann_dist, x_axis, y_axis):
+        energy_transfer = np.linspace(y_axis.end, y_axis.start, self._get_number_of_steps(y_axis))
+        momentum_transfer = np.linspace(x_axis.start, x_axis.end, self._get_number_of_steps(x_axis))
+        momentum_transfer = np.square(momentum_transfer[:scattering_data.shape[0]])
+        gdos = scattering_data * momentum_transfer[:,None]
+        gdos = gdos * energy_transfer[:,None]
```

Comment:
I think the following operations could use `*=` to do them in place for the same reason as above.